src/MongoDB/Manager.c

```diff
@@ -200,6 +200,15 @@ static void php_phongo_manager_prep_uri_options(zval *options TSRMLS_DC) /* {{{
 				continue;
 			}
 
+            if (!strcasecmp(ZSTR_VAL(string_key), MONGOC_URI_READPREFERENCE)) {
+				ZVAL_DEREF(option);
+				SEPARATE_ZVAL_NOREF(option);
+				if (Z_TYPE_P(option) != IS_STRING) {
+                    phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Unsupported readPreference type, a string must be given");
```

Comment:
Since you `continue` iteration below, it's possible for several exceptions to be thrown if other validations are added in the future. This would require users to rely on `Exception::getPrevious()` to view them all, as only the most recently thrown exception will be immediately accessible.

By convention, we'd want to throw an exception and return immediately with a failure code. Additionally, we'd check for a failure in the calling context (i.e. `Manager::__construct()`) and return there as well, so that we don't proceed with trying to initialize the object any further. Note that `php_phongo_manager_prep_uri_options()` is `void`, so that would need to be changed. If you look at `php_phongo_manager_merge_context_options()` and how it's called from the constructor, you'll see an example of what I'm describing.

Validating types may be outside of the purpose of `php_phongo_manager_prep_authmechanismproperties()` and `php_phongo_manager_prep_uri_options()`, since these functions were created to convert PHP values for BSON encoding that will later happen in `phongo_manager_init()`.

I think it may be preferable to introduce validation in the `php_phongo.c` functions that actually apply the converted BSON values to the libmongoc URI struct. See [`php_phongo_apply_options_to_uri()`](https://github.com/mongodb/mongo-php-driver/blob/f8d0a3301d68ace9261db9d19ab9d4fcdf724984/php_phongo.c#L965) for the generic case:

```
if (mongoc_uri_option_is_int32(key) && BSON_ITER_HOLDS_INT32(&iter)) {
	if (!mongoc_uri_set_option_as_int32(uri, key, bson_iter_int32(&iter))) {
		phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Failed to parse \"%s\" URI option", key);
		return false;
	}

	continue;
}
```

We currently ignore options unless the name indicates that libmongoc expects a given type _and_ the value matches that type. If we were to add validation here, we'd throw an exception if the BSON iterator did not hold the actual type.

----

Lastly, `Query.c` has numerous examples of exception messages for expected type errors. For consistency, we'd want to use something like:

```
phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Expected \"FOOBAR\" option to be string, %s given", zend_get_type_by_const(type));
```

If we're validating types while working with BSON, we can no longer rely on `zend_get_type_by_const()`; however, we can sort out error messages later.